src/go/languageservice/token_service.go

```diff
@@ -0,0 +1,157 @@
+package languageservice
+
+import (
+    "go/ast"
+    "go/token"
+)
+
+type tyNode interface {
```

Comment:
What's the benefit of the interfaces in this particular case given that they're all different? I think they're a helpful addition when they can be used to reduce boiler plate but in this particular case they don't seem to have.

Reply:
It's not so much about reducing boiler plate as it is about reducing dependencies on the token components. I.E. in another instance where you have to do something with many tokens but specific to each one, you just have to add that type of action as an interface method.